src/main/java/com/impactupgrade/nucleus/service/logic/ContactService.java

```diff
@@ -94,6 +97,24 @@ protected void fetchAndSetDonorData(PaymentGatewayEvent paymentGatewayEvent) thr
           contactSearch.keywords = Set.of(paymentGatewayEvent.getCrmAccount().billingAddress.street);
           existingContact = crmService.searchContacts(contactSearch).getSingleResult();
         }
+
```

Comment:
The below is the primary fix and a discussion about it. Everything else is purely adding a new method to CrmService (and a bit of cleanup).